### ISSUE TITLE
fix:  kubectl  karmada deinit  doesn't cascade deletion of pvc resources

### ISSUE THREAD

**Issue by CharlesQQ** [karmada-io/karmada#4750]
**What happened**:
1. Create karmada control plane as as the following command:
`kubectl-karmada init --kube-image-registry=registry.cn-hangzhou.aliyuncs.com/google_containers --storage-classes-name=local-path  --etcd-pvc-size='30Gi' --etcd-storage-mode='PVC'  --wait-component-ready-timeout=360 --etcd-replicas 3   --karmada-apiserver-replicas 3`
2. pod and pvc created successfully
![image](https://github.com/karmada-io/karmada/assets/18431718/e13fc21a-7c7a-4d6e-bc40-30754f5fafc0)
3. remove karmada control plane as  the following command:
`kubectl  karmada deinit`
4. pod deleted but pvc not
![image](https://github.com/karmada-io/karmada/assets/18431718/a8b7a70e-2042-4444-8aab-d816360bc54b)

**What you expected to happen**:

pvc should  be cascade deleted when etcd pods deleted.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version:   karmada 1.9
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by CharlesQQ** [comment]
After k8s 1.27,  support statefulset-pvc-auto-deletion,  refer to (https://kubernetes.io/blog/2023/05/04/kubernetes-1-27-statefulset-pvc-auto-deletion-beta/); but earlier version use must delete pvc manually.

**Comment by CharlesQQ** [comment]
Introduced flag  --etcd-sts-pvc-delete-policy=Delete to set field 
 sts.spec.persistentVolumeClaimRetentionPolicy.whenDeleted value

![image](https://github.com/karmada-io/karmada/assets/18431718/c02d464a-a133-4e1c-a0e0-f2adbbea2c9e)

**Comment by XiShanYongYe-Chang** [comment]
> I have a idea, Introduced flag --etcd-sts-pvc-delete-policy=Delete to set field `sts.spec.persistentVolumeClaimRetentionPolicy.whenDeleted` value

I agree with this idea, would you like to submit a PR? The flag name can be discussed in the PR

**Comment by CharlesQQ** [comment]
> I agree with this idea, would you like to submit a PR? The flag name can be discussed in the PR

sure!
